common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -314,6 +321,50 @@ public static String simpleClassName(Class<?> clazz) {
         }
     }
 
+    /**
+     * Quotes the specified value, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     * <p>
+     * The {@link CharSequence} will be quoted is if contains a
+     * {@value StringUtil#COMMA_CHAR},
+     * {@value StringUtil#NEWLINE_CHAR},
+     * {@value StringUtil#DOUBLE_QUOTE_CHAR},
+     *
+     * If a double quote is found inside of the value, it will be escaped by preceeding it
+     * with another double quote.
+     *
+     * @param value the value to be appended, and quoted if necessary
+     * @return {@link CharSequence} the quoted value
+     */
+    public static CharSequence quoteCsv(CharSequence value) {
```

Comment:
I think we need a function which is `escapeCsv`.  This function should do the following:

1. Escape all special characters that are not escaped (precede with `"` character).
2. Wrap value in quotes if any escapes where added or `,` character is found (and value is not already wrapped in quotes).